### PR TITLE
Alluxio ui updates

### PR DESCRIPTION
**Pull request by WilliamZapata**

### CODE REVIEW

File: alluxio-ui/worker/src/containers/pages/BlockInfo/BlockInfo.tsx [hunk 1]
```diff
@@ -64,14 +61,13 @@ class BlockInfo extends React.Component<AllProps, IBlockInfoState> {
...
-    const {refreshValue: prevRefreshValue, location: {search: prevSearch}} = prevProps;
+    const {refresh, location: {search}} = this.props;
+    const {refresh: prevRefresh, location: {search: prevSearch}} = prevProps;
     if (search !== prevSearch) {
       const {path, offset, limit, end} = parseQuerystring(search);
       this.setState({path, offset, limit, end});
       this.fetchData(path, offset, limit, end);
-    }
-    if (refreshValue !== prevRefreshValue) {
+    } else if (refresh !== prevRefresh) {
```
Comment (calvinjia) [inline alluxio-ui/worker/src/containers/pages/BlockInfo/BlockInfo.tsx]: With this logic, we can only update one `search` or `refresh`, is this intended?

File: alluxio-ui/common/src/App.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          This will contain reusable components for other web applications.
+        </p>
+        <p>
+          Control list:
+          <LoadingMessage/>
+        </p>
+      </div>
+    );
+  }
+}
```
Comment (calvinjia) [inline alluxio-ui/common/src/App.tsx]: Is this page and its contents necessary?

File: .gitignore [hunk 3]
```diff
@@ -48,3 +48,19 @@
...
+/alluxio-ui/**/**/lib/
+/alluxio-ui/**/**/node_modules
+/alluxio-ui/**/**/npm-debug.log
+/alluxio-ui/**/**/src/**/*.css
+/alluxio-ui/**/.sass-cache/
+/alluxio-ui/**/lerna-debug.log
+/alluxio-ui/**/node
+/alluxio-ui/**/node_modules
+/alluxio-ui/**/npm-debug.log
+/\?/.java/**/*
```
[the remark this answers is not in the document]
Comment (WilliamZapata) [reply .gitignore]: :/ not sure where this came from, removing

File: alluxio-ui/common/public/index.html [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      If you open it directly in the browser, you will see an empty page.
+
+      You can add webfonts, meta tags, or analytics to this file.
+      The build step will place the bundled scripts into the <body> tag.
+
+      To begin the development, run `npm start` or `yarn start`.
+      To create a production bundle, use `npm run build` or `yarn build`.
+    -->
+  </body>
+</html>
```
[the remark this answers is not in the document]
Comment (WilliamZapata) [reply alluxio-ui/common/public/index.html]: Yes, Really, all we use it for is a template for all web pages.

File: alluxio-ui/worker/src/containers/pages/BlockInfo/BlockInfo.tsx (hunk 1 above)
Comment (WilliamZapata) [reply alluxio-ui/worker/src/containers/pages/BlockInfo/BlockInfo.tsx]: Yes, I figure why sent two identical requests, the refresh one will request the same payload it had before, so whenever the search string changes and the refresh is triggered we will only send one.

File: alluxio-ui/common/src/App.tsx (hunk 2 above)
Comment (WilliamZapata) [reply alluxio-ui/common/src/App.tsx]: The common app is technically a buildable runnable product, so if we want to keep it that way, then yes.

File: alluxio-ui/common/src/components/FileView/FileView.tsx [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    } = this.props;
+
+    return (
+      <React.Fragment>
+        <h5>
+          {viewData.currentDirectory ? viewData.currentDirectory.absolutePath : viewData.currentPath}: <small>First 5KB
+          from {viewData.viewingOffset} in ASCII</small>
+        </h5>
+        <Form className="mb-3 viewData-file-form" id="viewDataFileForm" inline={true}>
+          <FormGroup className="mb-2 mr-sm-2 w-100">
```
Comment (calvinjia) [inline alluxio-ui/common/src/components/FileView/FileView.tsx]: Are these class names standard, and how do we decide which ones to use?

File: alluxio-ui/common/src/components/Header/Header.tsx [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+import {faCheckSquare, faSquare} from '@fortawesome/free-regular-svg-icons'
```
Comment (calvinjia) [inline alluxio-ui/common/src/components/Header/Header.tsx]: A general question about the new `tsx` files - is there usually any code documentation associated with them (similar to Javadocs)?

File: alluxio-ui/master/config-overrides.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  jest: function (config) {
+    // ...add your jest config customisation...
+    const commonUiModulesPath = resolveApp('../common/node_modules');
+    config.snapshotSerializers = ['enzyme-to-json/serializer'];
+    config.moduleDirectories = ['node_modules', commonUiModulesPath];
+    return config;
+  },
+  // The Webpack config to use when compiling your react app for development or production.
+  webpack: function (config, env) {
+    // ...add your webpack config customisation, rewires, etc...
```
Comment (calvinjia) [inline alluxio-ui/master/config-overrides.js]: Are these comments necessary?

File: alluxio-ui/master/public/index.html [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+      If you open it directly in the browser, you will see an empty page.
+
+      You can add webfonts, meta tags, or analytics to this file.
+      The build step will place the bundled scripts into the <body> tag.
+
+      To begin the development, run `npm start` or `yarn start`.
+      To create a production bundle, use `npm run build` or `yarn build`.
+    -->
+  </body>
+</html>
```
Comment (calvinjia) [inline alluxio-ui/master/public/index.html]: Is this page necessary / should it have different content?

File: alluxio-ui/master/src/constants/data/footerNavigation.tsx [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+  url: 'https://groups.google.com/forum/#!forum/alluxio-users'
+}, {
+  innerText: 'User Survey',
+  url: 'http://alluxio.org/resources/survey-users'
+}, {
+  innerText: 'Workers',
+  url: '/workers'
+}, {
+  innerText: 'Resources',
+  url: 'http://alluxio.org/alluxio-resources/'
```
Comment (calvinjia) [inline alluxio-ui/master/src/constants/data/footerNavigation.tsx]: Make the `http` urls `https`?

File: alluxio-ui/master/src/containers/App/App.test.tsx [hunk 10]
```diff
@@ -0,0 +1,11 @@
...
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
```
Comment (calvinjia) [inline alluxio-ui/master/src/containers/App/App.test.tsx]: Is this file necessary?

File: alluxio-ui/master/src/containers/pages/Overview/Overview.tsx [hunk 11]
```diff
@@ -0,0 +1,220 @@
...
+  }
+
+  public componentWillMount() {
+    this.props.fetchRequest();
+  }
+
+  public render() {
+    const {errors, overview} = this.props;
+
+    if (errors) {
```
Comment (calvinjia) [inline alluxio-ui/master/src/containers/pages/Overview/Overview.tsx]: If we are debugging this, where can we find the actual error?

File: alluxio-ui/master/src/containers/pages/Overview/Overview.tsx [hunk 12]
```diff
@@ -0,0 +1,220 @@
...
+                </tbody>
+              </Table>
+            </div>
+            <div className="col-md-6">
+              <h5>Cluster Usage Summary</h5>
+              <Table hover={true}>
+                <tbody>
+                <tr>
+                  <th scope="row">Workers Capacity</th>
+                  <td>{overview.freeCapacity}</td>
```
Comment (calvinjia) [inline alluxio-ui/master/src/containers/pages/Overview/Overview.tsx]: Is this field correct?

File: alluxio-ui/master/src/containers/pages/Overview/Overview.tsx [hunk 13]
```diff
@@ -0,0 +1,220 @@
...
+                  <th scope="row">Workers Free / Used</th>
+                  <td>{overview.freeCapacity} / {overview.usedCapacity}</td>
+                </tr>
+                <tr>
+                  <th scope="row">UnderFS Capacity</th>
+                  <td>{overview.diskCapacity}</td>
+                </tr>
+                <tr>
+                  <th scope="row">UnderFS Free / Used</th>
+                  <td>{overview.diskCapacity} / {overview.diskUsedCapacity}</td>
```
Comment (calvinjia) [inline alluxio-ui/master/src/containers/pages/Overview/Overview.tsx]: Is this field correct?

File: alluxio-ui/master/src/containers/pages/Workers/Workers.tsx [hunk 14]
```diff
@@ -0,0 +1,148 @@
...
+        <div className="container-fluid">
+          <div className="row">
+            <div className="col-12">
+              <h5>Live Workers</h5>
+              <Table hover={true}>
+                <thead>
+                <tr>
+                  <th>Node Name</th>
+                  <th>[D]Worker Id</th>
+                  <th>[D]Uptime</th>
```
Comment (calvinjia) [inline alluxio-ui/master/src/containers/pages/Workers/Workers.tsx]: These should only be displayed if debug is enabled.

File: alluxio-ui/master/src/containers/pages/Workers/Workers.tsx [hunk 15]
```diff
@@ -0,0 +1,148 @@
...
+                  <th>State</th>
+                  <th>Workers Capacity</th>
+                  <th>Space Used</th>
+                  <th>Space Usage</th>
+                </tr>
+                </thead>
+                <tbody>
+                {workers.normalNodeInfos.map((nodeInfo: INodeInfo) => (
+                  <tr key={nodeInfo.workerId}>
+                    <td><a href={`//${nodeInfo.host}:30000`} target="_blank">{nodeInfo.host}</a></td>
```
Comment (calvinjia) [inline alluxio-ui/master/src/containers/pages/Workers/Workers.tsx]: We shouldn't hard code `30000`

File: alluxio-ui/master/src/containers/pages/Workers/Workers.tsx [hunk 16]
```diff
@@ -0,0 +1,148 @@
...
+                    <td>{nodeInfo.state}</td>
+                    <td>{nodeInfo.capacity}</td>
+                    <td>{nodeInfo.usedMemory}</td>
+                    <td>
+                      <Progress multi={true}>
+                        <Progress bar={true} color="success"
+                                  value={`${nodeInfo.freeSpacePercent}`}>{nodeInfo.freeSpacePercent}% Free</Progress>
+                        <Progress bar={true} color="danger"
+                                  value={`${nodeInfo.usedSpacePercent}`}>{nodeInfo.usedSpacePercent}% Used</Progress>
+                      </Progress>
```
Comment (calvinjia) [inline alluxio-ui/master/src/containers/pages/Workers/Workers.tsx]: We use this component often, can we abstract it?

File: alluxio-ui/master/src/index.tsx [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+  if (action === 'POP') {
+    return;
+  }
+
+  // Allow the client to control scroll-to-top using location.state
+  if (loc.state && loc.state.scroll !== undefined && !loc.state.scroll) {
+    return;
+  }
+
+  // 200ms delay hack (for Firefox?)
```
Comment (calvinjia) [inline alluxio-ui/master/src/index.tsx]: Is this for firefox?

**Comment by WilliamZapata** [comment]
@calvin, I just realized I should be merging this into alluxio-ui, not master... will recreate.

File: alluxio-ui/master/src/store/browse/reducer.tsx [hunk 18]
```diff
@@ -0,0 +1,70 @@
...
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+import {Reducer} from 'redux';
+
+import {BrowseActionTypes, IBrowseState} from './types';
+
+const initialState: IBrowseState = {
```
Comment (calvinjia) [inline alluxio-ui/master/src/store/browse/reducer.tsx]: How does initial state work, would it be possible to default it without explicitly setting all the fields?

File: alluxio-ui/pom.xml [hunk 19]
```diff
@@ -0,0 +1,115 @@
...
+        <executions>
+          <execution>
+            <id>install node and npm</id>
+            <goals>
+              <goal>install-node-and-npm</goal>
+            </goals>
+            <phase>generate-sources</phase>
+            <configuration>
+              <nodeVersion>v10.11.0</nodeVersion>
+              <npmVersion>6.4.1</npmVersion>
```
Comment (calvinjia) [inline alluxio-ui/pom.xml]: Could you declare these versions as properties?

File: alluxio-ui/worker/src/constants/data/footerNavigation.tsx [hunk 20]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+import {INavigationData} from '@alluxio/common-ui/src/constants';
+
+export const footerNavigationData : INavigationData[] = [{
```
Comment (calvinjia) [inline alluxio-ui/worker/src/constants/data/footerNavigation.tsx]: Is this footer the same as the other one?

File: alluxio-ui/worker/src/constants/data/footerNavigation.tsx [hunk 21]
```diff
@@ -0,0 +1,29 @@
...
+  url: 'http://alluxio.org/'
+}, {
+  innerText: 'User Mailing List',
+  url: 'https://groups.google.com/forum/#!forum/alluxio-users'
+}, {
+  innerText: 'User Survey',
+  url: 'http://alluxio.org/resources/survey-users'
+}, {
+  innerText: 'Workers',
+  url: '/workers'
```
Comment (calvinjia) [inline alluxio-ui/worker/src/constants/data/footerNavigation.tsx]: Does this url work from the worker ui?

File: alluxio-ui/worker/src/constants/data/headerNavigation.tsx [hunk 22]
```diff
@@ -0,0 +1,29 @@
...
+  url: '/blockInfo'
+}, {
+  innerText: 'Logs',
+  url: '/logs'
+}, {
+  innerText: 'Metrics',
+  url: '/metrics'
+}, {
+  innerText: 'Return to Master',
+  url: () => `${window.location.origin.replace(/\:[0-9]+$/, ':19999')}`
```
Comment (calvinjia) [inline alluxio-ui/worker/src/constants/data/headerNavigation.tsx]: We shouldn't hard code 19999? Also, is this sufficient, we need to change to the master address right?

File: alluxio-ui/common/src/components/FileView/FileView.tsx (hunk 5 above)
Comment (WilliamZapata) [reply alluxio-ui/common/src/components/FileView/FileView.tsx]: Yes, these are part of the bootstrap framework

File: alluxio-ui/common/src/components/Header/Header.tsx (hunk 6 above)
Comment (WilliamZapata) [reply alluxio-ui/common/src/components/Header/Header.tsx]: Yes, we can add this (https://github.com/Microsoft/TypeScript/wiki/JSDoc-support-in-JavaScript).

File: alluxio-ui/master/config-overrides.js (hunk 7 above)
Comment (WilliamZapata) [reply alluxio-ui/master/config-overrides.js]: I'll remove them

File: alluxio-ui/master/public/index.html (hunk 8 above)
Comment (WilliamZapata) [reply alluxio-ui/master/public/index.html]: This is a template page used by create-react-app.

File: alluxio-ui/master/src/constants/data/footerNavigation.tsx (hunk 9 above)
Comment (WilliamZapata) [reply alluxio-ui/master/src/constants/data/footerNavigation.tsx]: Good point, although it should redirect.

File: alluxio-ui/master/src/containers/App/App.test.tsx (hunk 10 above)
Comment (WilliamZapata) [reply alluxio-ui/master/src/containers/App/App.test.tsx]: That's where the test for App.tsx will be. I like to put test files alongside the thing they test instead of in a separate tests directory. I think it makes things easier in the long run.

File: alluxio-ui/master/src/containers/pages/Metrics/Metrics.tsx [hunk 23]
```diff
@@ -0,0 +1,323 @@
...
+    if (errors) {
+      return (
+        <Alert color="danger">
+          Unable to reach the api endpoint for this page.
+        </Alert>
+      );
+    }
+
+    return (
+      <div className="metrics-page">
```
[the remark this answers is not in the document]
Comment (WilliamZapata) [reply alluxio-ui/master/src/containers/pages/Metrics/Metrics.tsx]: Yes, I'll make a note.

File: alluxio-ui/master/src/containers/pages/Overview/Overview.tsx [hunk 24]
```diff
@@ -0,0 +1,220 @@
...
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+import {AxiosResponse} from 'axios';
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (WilliamZapata) [reply alluxio-ui/master/src/containers/pages/Overview/Overview.tsx]: Yes, one is exporting a named module, the other is exporting the default module. It depends on the library and what I want from it.

File: alluxio-ui/master/src/containers/pages/Overview/Overview.tsx (hunk 11 above)
Comment (WilliamZapata) [reply alluxio-ui/master/src/containers/pages/Overview/Overview.tsx]: Errors will have the actual response object from the api. It will be within this object.

File: alluxio-ui/master/src/containers/pages/Workers/Workers.tsx (hunk 14 above)
Comment (WilliamZapata) [reply alluxio-ui/master/src/containers/pages/Workers/Workers.tsx]: Good point, I need a new api to support this.

File: alluxio-ui/master/src/containers/pages/Workers/Workers.tsx (hunk 15 above)
Comment (WilliamZapata) [reply alluxio-ui/master/src/containers/pages/Workers/Workers.tsx]: I need to add this to a new init api.

File: alluxio-ui/master/src/containers/pages/Workers/Workers.tsx (hunk 16 above)
Comment (WilliamZapata) [reply alluxio-ui/master/src/containers/pages/Workers/Workers.tsx]: <Progress/> is the abstracted component.

File: alluxio-ui/master/src/index.tsx (hunk 17 above)
Comment (WilliamZapata) [reply alluxio-ui/master/src/index.tsx]: yep, it's a browser workaround

File: alluxio-ui/master/src/store/browse/reducer.tsx (hunk 18 above)
Comment (WilliamZapata) [reply alluxio-ui/master/src/store/browse/reducer.tsx]: What do you mean default it... the default state is what appears in the store before we make any api calls.

File: alluxio-ui/pom.xml (hunk 19 above)
Comment (WilliamZapata) [reply alluxio-ui/pom.xml]: Maybe, but why? I think we might want to manually change these since they affect the UI node engine.

File: alluxio-ui/worker/src/constants/data/footerNavigation.tsx (hunk 20 above)
Comment (WilliamZapata) [reply alluxio-ui/worker/src/constants/data/footerNavigation.tsx]: I think yes, if so I'll move it to common.

File: alluxio-ui/worker/src/constants/data/footerNavigation.tsx (hunk 21 above)
Comment (WilliamZapata) [reply alluxio-ui/worker/src/constants/data/footerNavigation.tsx]: :( I'll fix

File: alluxio-ui/worker/src/constants/data/headerNavigation.tsx (hunk 22 above)
Comment (WilliamZapata) [reply alluxio-ui/worker/src/constants/data/headerNavigation.tsx]: You're right, I need a way of knowing where the master is... this works where the master and worker are in the same machine. I'll fix this.

File: alluxio-ui/master/src/store/browse/reducer.tsx (hunk 18 above)
Comment (calvinjia) [reply alluxio-ui/master/src/store/browse/reducer.tsx]: There is a lot boilerplate here, I was wondering if there was a way to get around that.

File: alluxio-ui/pom.xml (hunk 19 above)
Comment (calvinjia) [reply alluxio-ui/pom.xml]: Usually we have a properties section in the pom so it is easier to see where all the version are and if we depend on the same thing in multiple modules we use the same version.

File: alluxio-ui/master/src/containers/pages/Overview/Overview.tsx (hunk 12 above)
Comment (WilliamZapata) [reply alluxio-ui/master/src/containers/pages/Overview/Overview.tsx]: no, I will fix this

File: alluxio-ui/master/src/containers/pages/Overview/Overview.tsx (hunk 13 above)
Comment (WilliamZapata) [reply alluxio-ui/master/src/containers/pages/Overview/Overview.tsx]: I will also fix this

File: alluxio-ui/master/src/containers/pages/Workers/Workers.tsx (hunk 16 above)
Comment (WilliamZapata) [reply alluxio-ui/master/src/containers/pages/Workers/Workers.tsx]: I meant to say the `<Progress/>` component is the abstracted one, there are two here.
